### PR TITLE
x64: expand FloatCC enum in ISLE

### PR DESCRIPTION
**Pull request by abrown**
To lower the various `select` rules in ISLE, we need visibility of the `FloatCC` enum variants, not just a `primitive` type. This change generates the ISLE `enum` for `FloatCC` in `clif.isle`.

### CODE REVIEW

File: cranelift/codegen/meta/src/gen_inst.rs [hunk 1]
```diff
@@ -1104,20 +1107,42 @@ fn gen_isle(formats: &[&InstructionFormat], instructions: &AllInstructions, fmt:
...
                 .collect::<Vec<_>>()
         })
         .collect();
-    for ty in imm_tys {
+    for ty in imm_tys.keys().filter(|&&k| k != "FloatCC") {
         fmtln!(fmt, "(type {} (primitive {}))", ty, ty);
     }
     fmt.empty_line();
 
+    // Generate `FloatCC` enumeration.
```
Comment (bjorn3) [inline cranelift/codegen/meta/src/gen_inst.rs:1126]: Can you do this for all enum operands including `IntCC`?

Comment (abrown) [reply cranelift/codegen/meta/src/gen_inst.rs:1126]: So `IntCC`, `AtomicRmwOp`, and `TrapCode` from [immediates.rs](https://github.com/bytecodealliance/wasmtime/blob/a7aa1723e0a00ab48a2cff304a8a8aa8b351c390/cranelift/codegen/meta/src/shared/immediates.rs#L160)?

Comment (bjorn3) [reply cranelift/codegen/meta/src/gen_inst.rs:1126]: Yes

**Review by cfallin — APPROVED** [review]
LGTM, thanks!

The dead-code warnings-as-errors in CI are a little annoying -- we could either put `#[cfg(feature = "rebuild-isle")]` on more functions, or just `#[allow(dead_code)]`. The former is a little haphazard; I'd prefer the latter with a comment indicating "actually used with rebuild-isle feature". Good to merge once that's fixed!

**Pull request merged**